tests/codeception/functional/Paragraphs/WYSIWYGCest.php

```diff
@@ -0,0 +1,194 @@
+    $I->canSee('Big Button', 'a.su-button--big');
+    $I->canSee('Secondary Button', 'a.su-button--secondary');
+    $I->canSee('emphasis', 'em');
+    $I->canSee('important', 'strong');
+    $I->canSeeNumberOfElements('blockquote', 1);
+    $I->cantSeeElement('.su-page-components footer');
+    $I->canSeeNumberOfElements('code', 2);
+    $I->canSeeNumberOfElements('dl', 1);
+    $I->canSeeNumberOfElements('dt', 2);
+    $I->canSeeNumberOfElements('dd', 2);
+
+    # List Tags
+    $I->canSee('This is a list', 'ul li');
+    $I->canSee('child list items', 'ul ul li');
+    $I->canSee('Ordered list item', 'ol li');
+    $I->canSee('Child ordered list item', 'ol ol li');
+
+    # Table Tags
+    $I->canSeeElement('table');
+    $I->canSeeNumberOfElements('caption', 1);
+    $I->canSeeNumberOfElements('caption', 1);
+    $I->canSeeNumberOfElements('tbody', 1);
+    $I->canSeeNumberOfElements('tr', 3);
+    $I->canSeeNumberOfElements('th[scope]', 2);
+    $I->canSeeNumberOfElements('td', 4);
+  }
+
+  /**
+   * Images in the WYSIWYG should display correctly.
+   */
+  public function testEmbeddedImage(FunctionalTester $I) {
+    $node = $this->getNodeWithParagraph($I, 'Lorem Ipsum');
+    $I->logInWithRole('administrator');
+    $I->amOnPage($node->toUrl()->toString());
+    $I->cantSeeElement('.su-page-components img');
+    $I->click('Edit', '.local-tasks-block');
+    $I->waitForElementVisible('#row-0');
+    $I->click('Edit', '.inner-row-wrapper');
+    $I->waitForElementVisible('.cke_inner');
+    $I->click('Insert from Media Library');
+    $I->waitForElementVisible('.dropzone');
+    $I->dropFileInDropzone(__DIR__ . '/logo.jpg');
+    $I->click('Upload and Continue');
+    $I->waitForText('Alternative text');
+    $I->clickWithLeftButton(".ui-dialog-buttonset button:nth-child(2)");
+    $I->waitForAjaxToFinish();
+    $I->click('Continue');
+    $I->waitForElementNotVisible('.MuiDialog-scrollPaper');
+    $I->click('Save');
+    $I->canSeeElement('.su-page-components img');
+  }
+
+  /**
+   * Videos in the WYSIWYG should display correctly.
+   */
+  public function testEmbeddedVideo(FunctionalTester $I) {
+    $node = $this->getNodeWithParagraph($I, 'Lorem Ipsum');
+    $I->logInWithRole('administrator');
+    $I->amOnPage($node->toUrl()->toString());
+    $I->cantSeeElement('iframe');
+    $I->click('Edit', '.local-tasks-block');
+    $I->waitForElementVisible('#row-0');
+    $I->click('Edit', '.inner-row-wrapper');
+    $I->waitForElementVisible('.cke_inner');
+    $I->click('Insert from Media Library');
+    $I->waitForElementVisible('.dropzone');
+    $I->click('Video', '.media-library-menu-video');
+    $I->waitForElementVisible('.media-library-add-form-oembed-url');
+    $I->fillField('Add Video via URL', 'https://www.youtube.com/watch?v=ktCgVopf7D0');
+    $I->click('Add');
+    $I->waitForText('The media item has been created but has not yet been saved');
+    $I->clickWithLeftButton(".ui-dialog-buttonset button:nth-child(2)");
```

Comment:
Does this have to explicitly be left button in codeception?

Reply:
I think I needed that at one point but I can change it now. If I remember, I was having race issues. But that should be solved now with the `waitForText` step above it.